### PR TITLE
Support Building Helm Charts From EKS-D Images

### PR DESCRIPTION
**Pull request by jonathanmeier5**
*Issue #, if available:*
Epic: https://github.com/aws/eks-anywhere/issues/3187
Ticket: https://github.com/aws/eks-anywhere/issues/3414

*Description of changes:*
- Adds functionality to automatically pull latest EKS-D project tag
- Adds functionality to skip metrics-server project, our first EKS-D backed helm chart, when generating UPSTREAM_PROJECTS.yaml

Supports https://github.com/aws/eks-anywhere-build-tooling/pull/1331.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: build/lib/eksd_releases.sh [hunk 1]
```diff
@@ -122,7 +122,7 @@ function build::eksd_releases::get_eksd_component_asset_path() {
 
     echo "$yaml" | yq e "$query" -
     
-    set -$oldopt
+    set -$(echo $oldopt | sed 's/c//') # remove -c from options for when this is invoked with bash -c
```
Comment (jaxesn) [inline build/lib/eksd_releases.sh:125]: Why did we just run into this? I ran into this on some unrelated bash change recently and didnt look close enough into why.

Comment (jonathanmeier5) [reply build/lib/eksd_releases.sh:125]: I believe it's because we're setting `-c` in our `shell` options [here](https://github.com/aws/eks-anywhere-build-tooling/blob/c6e0fc90e973ccca6c1dc4879f382bb57996ca66/Common.mk#L4-L4). It seems that once `-c` is set you can't reset or unset it in the way you can other flags, which makes sense given its function.

Comment (TerryHowe) [reply build/lib/eksd_releases.sh:125]: That is an odd argument for flags. It really isn't a flag

Comment (jonathanmeier5) [reply build/lib/eksd_releases.sh:125]: yeah, it's weird 😕

Comment (jonathanmeier5) [reply build/lib/eksd_releases.sh:125]: For posterity, these were the error messages:
```
/home/prow/go/src/github.com/aws/eks-anywhere-build-tooling/build/lib/eksd_releases.sh: line 37: set: -c: invalid option
set: usage: set [-abefhkmnptuvxBCHP] [-o option-name] [--] [arg ...]
/home/prow/go/src/github.com/aws/eks-anywhere-build-tooling/build/lib/eksd_releases.sh: line 125: set: -c: invalid option
```
